bofire/strategies/random.py

```diff
@@ -21,25 +23,52 @@ def __init__(
         **kwargs,
     ):
         super().__init__(data_model=data_model, **kwargs)
-        self._init_sampler()
-
-    def _init_sampler(self) -> None:
-        errors = []
-        for dm, functional in [
-            (data_models.PolytopeSampler, PolytopeSampler),
-            (data_models.RejectionSampler, RejectionSampler),
-        ]:
-            try:
-                data_model = dm(domain=self.domain)
-                self.sampler = functional(data_model=data_model)
-                break
-            except ValidationError as err:
-                errors.append(err)
-        if self.sampler is None:
-            raise Exception(errors)
+        self.num_base_samples = data_model.num_base_samples
+        self.max_iters = data_model.max_iters
+        self.fallback_sampling_method = data_model.fallback_sampling_method
+        self.n_burnin = data_model.n_burnin
+        self.n_thinning = data_model.n_thinning
+
+    def _get_feasible_domain_for_polytope_sampler(self) -> Domain:
```

Comment:
Make this a free function with `domain: Domain` as input parameter for better encapsulation. It does not need access to all of `self`.

Reply:
Done.